docs/poisoning.md

```diff
@@ -0,0 +1,167 @@
+
+
+## Threat Models
+
+There are currently three threat models handled by Armory: dirty-label backdoor, clean-label backdoor, and Witches' Brew (clean-label gradient matching).  In a backdoor attack, an adversary adds a small trigger, or backdoor, to a small portion of the train set in order to gain control of the the model at test time.
+The trigger is usually a small (but not imperceptible) image superposed on the data, and the adversary's goal is to force the model to misclassify test images that have the trigger applied.  Armory includes several trigger images under `utils/triggers/`.
+
+
+In poisoning attacks, the term _source class_ refers to the label of the image(s) that the adversary hopes to misclassify.  In the case of a targeted attack, the _target class_ is the desired misclassification label.  Neither of these terms describes which class gets poisoned; that depends on the threat model.  All of Armory's poisoning scenarios perform targeted attacks.  For simplicity, most Armory scenarios assume a single source class (all images to misclassify are from the same class) and a single target class (all misclassifications are aiming for the same label).  The exception is Witches' Brew, which accepts images from arbitrary source classes that can all have distinct targets.
+
+
+### Dirty-label backdoor
+
+In a [Dirty-label Backdoor (DLBD) Attack](https://arxiv.org/abs/1708.06733), training images are chosen from the source class, have a trigger applied to them, and then have their labels flipped to the target class.  The model is then trained on this modified data.  The adversary's goal is that test images from the source class will be classified as `target` when the trigger is applied at test time.
+
+
+
+### Clean-label backdoor
+
+In a [Clean-label Backdoor (CLBD) Attack](https://people.csail.mit.edu/madry/lab/cleanlabel.pdf), the triggers are applied to target-class images during training.  The poisoned samples also undergo some imperceptible gradient-based modifications to weaken the natural features, thus strengthening the association of the trigger with the target class label.
+At test time, the adversary applies the trigger to source-class images in order to get them to misclassify as `target`.
+
+
+
+### Witches' brew
+
+[Witches' Brew](https://arxiv.org/abs/2009.02276) is a clean-label attack but there is no backdoor or trigger involved.  The adversary selects individual `source` images from the test set; these are the images that the adversary wants to misclassify as `target` and are called _triggers_, not to be confused with the backdoor trigger described before.  The attack uses a gradient-matching algorithm to modify a portion of the train-set target class, such that the unmodified test-set triggers will be misclassified.
```

Comment:
> The attack uses a gradient-matching algorithm to modify a portion of the train-set target class, such that the unmodified test-set triggers will be misclassified.

Is poisoning only members of the target class part of the attack or just our implementation?

Reply:
It's Taesung's (ART's) implementation.  Not sure whether the original formulation required it.